orderapi/src/main/java/fi/hel/verkkokauppa/order/service/accounting/AccountingSlipService.java

```diff
@@ -285,11 +299,57 @@ private Map<String, List<OrderItemAccountingDto>> groupByCompanyCode(List<OrderI
         return groupedAccountings;
     }
 
+    private Map<String, List<OrderItemAccountingDto>> groupOrderItemAccountingsByBalanceProfitCenter(List<OrderItemAccountingDto> accountingItemsForDate) {
+        Map<String, List<OrderItemAccountingDto>> groupedAccountings = new HashMap<>();
+
+        for (OrderItemAccountingDto orderItemAccountingDto : accountingItemsForDate) {
+            String balanceProfitCenter = orderItemAccountingDto.getBalanceProfitCenter();
+            List<OrderItemAccountingDto> accountingList = groupedAccountings.get(balanceProfitCenter);
+
+            if (accountingList == null) {
+                ArrayList<OrderItemAccountingDto> accountings = new ArrayList<>();
+                accountings.add(orderItemAccountingDto);
+                groupedAccountings.put(balanceProfitCenter, accountings);
+            } else {
+                accountingList.add(orderItemAccountingDto);
+            }
+        }
+        return groupedAccountings;
+    }
+
+    public Map<String, List<AccountingSlipRowDto>> groupAccountingSlipRowsByBalanceProfitCenter(List<AccountingSlipRowDto> rows) {
+        Map<String, List<AccountingSlipRowDto>> groupedAccountings = new HashMap<>();
+
+        for (AccountingSlipRowDto row : rows) {
+            String balanceProfitCenter = row.getBalanceProfitCenter();
+            List<AccountingSlipRowDto> accountingList = groupedAccountings.get(balanceProfitCenter);
+
+            if (accountingList == null) {
+                ArrayList<AccountingSlipRowDto> accountings = new ArrayList<>();
+                accountings.add(row);
+                groupedAccountings.put(balanceProfitCenter, accountings);
+            } else {
+                accountingList.add(row);
+            }
+        }
+        return groupedAccountings;
+    }
+
     private AccountingSlip createAccountingAndRows(AccountingSlipDto accountingSlipDto) {
         AccountingSlip accountingSlip = accountingSlipRepository.save(new AccountingSlipTransformer().transformToEntity(accountingSlipDto));
 
         List<AccountingSlipRowDto> rows = accountingSlipDto.getRows();
-        rows.forEach(row -> accountingSlipRowRepository.save(new AccountingSlipRowTransformer().transformToEntity(row)));
+
+        rows.forEach(row -> {
+            // do not save vat or income entry rows
```

Comment:
Why we dont want to save vat or income entry rows?

Reply:
Just because it hadn't been done before. Logic that re-exports already exported only expects to have sum rows. I was considering saving them but would have needed other kind of changes to make sure all rows are in correct order as in quick test they changed order (income entry rows and vat rows were not in correct place). Possibly would have needed new client/service for reading the rows and row number added to the index for sorting. I just went with the first solution I was already making because of the deadline